### PR TITLE
Added support of 64bit events.

### PR DESCRIPTION
**Pull request by Hadatko**
Signed-off-by: Cervenka Dusan <cervenka@acrios.com>

Hello, i am curious if i can add support for more events than 24. It looks like people are running out of current highest scope. 

-----------

I added macro condtions for configUSE_16_BIT_TICKS = 2 in which case 64bit value will be used. 
-----------

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by jasonpcarroll** [comment]
Hi @Hadatko,
I will bring this PR up with the team. My immediate suggestion would be to make a new config called configUSE_64_BIT_TICKS, instead of using a higher value for configUSE_16_BIT_TICKS as that is a bit confusing, and then just check if configUSE_64_BIT_TICKS == 1.

Best,

Jason Carroll

**Comment by Hadatko** [comment]
Hi @jasonpcarroll , sounds great, but in that case i think it would be better if new name can cover all three cases and old one can be obsolote and of course check if both macros value are compatible.

**Comment by jasonpcarroll** [comment]
Funny - I was just thinking that as your replied and editting my comment above. :)

**Comment by Hadatko** [comment]
I am working same way. First is idea which will work. Then improved one, then improved one .... ;) Iteration to better quality. I am happy that we are on same line here ;)

**Comment by Hadatko** [comment]
When you have final decision let me know and maybe where to define these values: TICK_BIT_WIDTH_16, TICK_BIT_WIDTH_32, TICK_BIT_WIDTH_64 

and i can modify accordingly

File: event_groups.c [hunk 1]
```diff
@@ -54,6 +54,11 @@
     #define eventUNBLOCKED_DUE_TO_BIT_SET    0x0200U
     #define eventWAIT_FOR_ALL_BITS           0x0400U
     #define eventEVENT_BITS_CONTROL_BYTES    0xff00U
+#elif configUSE_16_BIT_TICKS == 2
+    #define eventCLEAR_EVENTS_ON_EXIT_BIT    0x0100000000000000UL
```
Comment (tcpluess) [inline event_groups.c]: I would suggest to use `ULL` for the 64 bit constants instead of just `UL`.

**Comment by Hadatko** [comment]
@jasonpcarroll Any news here?

**Comment by shubnil** [comment]
Hi, We are discussing this internally and we will get back soon. Thanks for being patient.

**Comment by Hadatko** [comment]
Ok no problem. Sorry for being not patient :D I was just curious if you noticed the changes ;) Happy Christmass ;)

**Comment by Hadatko** [comment]
Hello, Just pinging ;) Still no decision? :D

**Comment by aggarg** [comment]
Apologies for slipping. Taking a look.

File: include/FreeRTOS.h [hunk 2]
```diff
@@ -155,8 +155,41 @@
     #error Missing definition:  configUSE_TICK_HOOK must be defined in FreeRTOSConfig.h as either 1 or 0.  See the Configuration section of the FreeRTOS API documentation for details.
 #endif
 
-#ifndef configUSE_16_BIT_TICKS
-    #error Missing definition:  configUSE_16_BIT_TICKS must be defined in FreeRTOSConfig.h as either 1 or 0.  See the Configuration section of the FreeRTOS API documentation for details.
+#ifdef configUSE_16_BIT_TICKS
```
[the remark this answers is not in the document]
Comment (Hadatko) [reply include/FreeRTOS.h]: Hi, i moved branch up and added commit based on your requests. Hopefully is everything right now.

**Comment by aggarg** [comment]
Thank you for these changes. There are some compilation failures - can you look into those?

**Comment by Hadatko** [comment]
Hi @aggarg, my updated code is based on your suggestion. 
The issue with compilation looks like wrong suggested format:
image
How to rewrite it to make happy you and your validator?

**Review by n9wxu — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Comment by Hadatko** [comment]
Thank you ;)

**Pull request merged**